tests/clusters/test_cluster.py

```diff
@@ -196,3 +218,20 @@ def test_cache_connections() -> None:
     assert cluster_1.get_query_connection(
         cluster.ClickhouseClientSettings.QUERY
     ) != cluster_3.get_query_connection(cluster.ClickhouseClientSettings.QUERY)
+
+
+@patch("snuba.settings.OVERRIDE_STORAGE_SET_KEYS", OVERRIDE_STORAGE_SET_KEY_CONFIG)
+@patch("snuba.settings.CLUSTERS", TEST_SLICE_CONFIG)
+@patch(
+    "snuba.clusters.storage_sets._REGISTERED_STORAGE_SET_KEYS",
+    {"GENERIC_METRICS_DISTRIBUTIONS_1": "generic_metrics_distributions_1"},
+)
+def test_override_storage_set_keys() -> None:
+    importlib.reload(cluster)
+
+    res_cluster = cluster.get_cluster(StorageSetKey.GENERIC_METRICS_DISTRIBUTIONS, 1)
```

Comment:
Can you also add a test which asserts that without the optional slice, we map to the original cluster?